server/models/services/ModelService.js

```diff
@@ -0,0 +1,216 @@
+
+    // findByPk
+    return await model.findOne({
+        where: { id },
+        attributes,
+        include
+    });
+};
+
+/**
+ * @param {Model} model - Sequelize Model instance to query for
+ * @param {object} queryParams - values to be used to query Sequelize Model
+ * @param {any[]} attributes - attributes of the Sequelize Model to be returned
+ * @param {any[]} include - information on Sequelize Model relationships
+ * @returns {Promise<Model>} - Sequelize Model
+ */
+const getByQuery = async (
+    model,
+    queryParams,
+    attributes = [],
+    include = []
+) => {
+    if (!model) throw new Error('Model not defined'); // TODO: pass through custom APIError
+
+    return await model.findOne({
+        where: { ...queryParams },
+        attributes,
+        include
+    });
+};
+
+/**
+ * @link {https://sequelize.org/v5/manual/models-usage.html#-code-findandcountall--code----search-for-multiple-elements-in-the-database--returns-both-data-and-total-count}
+ * @param {Model} model - Sequelize Model instance to query for
+ * @param {object} filter - values to be used to search Sequelize Model
+ * @param {any[]} attributes - attributes of the Sequelize Model to be returned
+ * @param {any[]} include - information on Sequelize Model relationships
+ * @param {object} pagination - pagination options for query
+ * @param {number} [pagination.page=0] - page to be queried in the pagination result (affected by {@param pagination.enable})
+ * @param {number} [pagination.limit=10] - amount of results to be returned per page (affected by {@param pagination.enable})
+ * @param {string[][]} [pagination.order=[]] - expects a Sequelize structured input dataset for sorting the Sequelize Model results (NOT affected by {@param pagination.enable}). See {@link https://sequelize.org/v5/manual/querying.html#ordering} and {@example [ [ 'username', 'DESC' ], [..., ...], ... ]}
+ * @param {boolean} [pagination.enable=false] - use to enable pagination for a query result as well useful values. Data for all items matching query if not enabled
+ * @returns {Promise<*>} - collection of queried Sequelize Models or paginated structure if pagination flag is enabled
+ */
+const get = async (
+    model,
+    filter = {}, // pass to 'where' for top level model
+    attributes = [],
+    include = [],
+    pagination = {}
+) => {
+    if (!model) throw new Error('Model not defined'); // TODO: pass through custom APIError
+
+    // search and filtering options
+    let where = { ...filter };
+
+    // pagination and sorting options
+    let { page = 0, limit = 10, order = [], enable = false } = pagination; // page 0->1, 1->2; manage through middleware
+    // 'order' structure eg. [ [ 'username', 'DESC' ], [..., ...], ... ]
+    if (page < 0) page = 0;
+    if (limit < 0 || !enable) limit = null;
+    const offset = limit < 0 ? 0 : page * limit; // skip (1 * 10 results) = 10 to get get to page 2
```

Comment:
I don't think we need to check for negative numbers - if someone passes in a negative number and it explodes, it seems ok to me. That might make the code here slightly cleaner too! If you do think negative numbers are possible I think throwing an error would be appropriate.

Reply:
But it shouldn't explode! If it were to blow up, that's a potentially bad experience for the user and I support it because I'd rather default the calling method to the correct approach but throwing an error is ok as well.

Reply:
I get this - some tolerance to failure at such an important and frontend-facing feature makes sense.